### DISCUSSION TITLE
Sort order of CompletionItems display

### DISCUSSION THREAD

**Discussion by wsande** [python-cmd2/cmd2#1154] [Q&A] [answered]
If I have a list of CompletionItems like this:
[(1, 'red'), (2, 'blue'), (3, 'green'), ... (10, 'magenta'), (11, 'cyan')]

When I tab-complete the command, it displays the completion items like this:
```
ID
==============
1     red
10    magenta
11    cyan
2     blue
3     green
```
How do I get them to display in numerical order of item ID's instead of alphabetical order?

**Comment by kmvanbrunt** [comment, marked as answer]
First sort the results how you want them to display. Then set `self.matches_sorted = True` before returning them.

Here is a description of `matches_sorted` from cmd2.py. This value gets reset to `False` each time tab completion runs, so you don't need to reset it manually.

```Python
# Set to True before returning matches to complete() in cases where matches have already been sorted.
# If False, then complete() will sort the matches using self.default_sort_key before they are displayed.
# This does not affect self.formatted_completions.
self.matches_sorted = False
```

One thing to note is that `CompletionItem` inherits from string. Therefore the IDs are being converted to strings when you create each `CompletionItem`. To sort strings in numerical order, use a natural sort. **cmd2** provides natural sort functions in utils.py.

You can do either of the following:
```Python
from cmd2 import utils

# 1. This way sorts in place
my_list.sort(key=utils.natural_keys)

# 2. This way makes a copy of the list
my_list = utils.natural_sort(my_list)
```

If you sort the ID numbers before creating your `CompletionItem` list, then you won't need to use the natural sort functions.

**Reply by wsande** [reply]
Thanks!  That's exactly what I needed
